pyuvsim/simsetup.py

```diff
@@ -502,39 +514,94 @@ def parse_telescope_params(tele_params, config_path):
             |  antenna_numbers: corresponding list of antenna numbers
             |  antenna_positions: Array of ECEF antenna positions
             |  telescope_location: ECEF array center location
+            |  telescope_location_lat_lon_alt: Lat Lon Alt array center location
             |  telescope_config_file: Path to configuration yaml file
             |  antenna_location_file: Path to csv layout file
             |  telescope_name: observatory name
         beam_list:  Beam models in the configuration.
         beam_dict:  Antenna numbers to beam indices
     """
-
-    telescope_config_name = tele_params['telescope_config_name']
-    layout_csv = tele_params['array_layout']
-    if not os.path.isdir(config_path):
-        config_path = os.path.dirname(config_path)
+    tele_params = copy.deepcopy(tele_params)
+    # check for telescope config
+    tele_config = 'telescope_config_name' in tele_params
+    if tele_config:
+        # parse telescope config
         if not os.path.isdir(config_path):
-            raise ValueError('config_path from yaml is not a directory')
-    if not os.path.exists(telescope_config_name):
-        telescope_config_name = os.path.join(config_path, telescope_config_name)
+            config_path = os.path.dirname(config_path)
+            if not os.path.isdir(config_path):
+                raise ValueError('config_path {} is not a directory'.format(config_path))
+        telescope_config_name = tele_params['telescope_config_name']
         if not os.path.exists(telescope_config_name):
-            raise ValueError('telescope_config_name file from yaml does not exist')
-    if not os.path.exists(layout_csv):
-        layout_csv = os.path.join(config_path, layout_csv)
+            telescope_config_name = os.path.join(config_path, telescope_config_name)
+            if not os.path.exists(telescope_config_name):
+                raise ValueError('telescope_config_name file from yaml does not exist')
+        with open(telescope_config_name, 'r') as yf:
+            telconfig = yaml.safe_load(yf)
+            telescope_location_latlonalt = ast.literal_eval(telconfig['telescope_location'])
+            telescope_location = list(telescope_location_latlonalt)
+            telescope_location[0] *= np.pi / 180.
+            telescope_location[1] *= np.pi / 180.   # Convert to radians
+            tele_params['telescope_location'] = uvutils.XYZ_from_LatLonAlt(*telescope_location)
+            telescope_name = telconfig['telescope_name']
+
+    else:
+        # if not provided, get bare-minumum keys from tele_params
+        if 'telescope_location' not in tele_params:
+            raise KeyError("If telescope_config_name not provided in tele_params, must provide telescope_location")
```

Comment:
Since `tele_params` is an internally-used variable name, this should probably be rephrased to something clearer to the user.